instrumentation-genai/opentelemetry-instrumentation-vertexai-v2/tests/conftest.py

```diff
@@ -0,0 +1,99 @@
+"""Unit tests configuration module."""
+
+import re
+from typing import Any, Mapping, MutableMapping
+
+import pytest
+import vertexai
+from google.auth.credentials import AnonymousCredentials
+from vcr import VCR
+from vcr.record_mode import RecordMode
+from vcr.request import Request
+
+from opentelemetry import trace
+from opentelemetry.instrumentation.vertexai_v2 import VertexAIInstrumentor
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import SimpleSpanProcessor
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import (
+    InMemorySpanExporter,
+)
+
+pytest_plugins = []
+
+FAKE_PROJECT = "fake-project"
+
+
+@pytest.fixture(scope="session")
+def exporter():
+    span_exporter = InMemorySpanExporter()
+    processor = SimpleSpanProcessor(span_exporter)
+
+    provider = TracerProvider()
+    provider.add_span_processor(processor)
+    trace.set_tracer_provider(provider)
+
+    VertexAIInstrumentor().instrument()
+
+    return span_exporter
+
+
+@pytest.fixture(autouse=True)
+def clear_exporter(exporter):  # pylint: disable=redefined-outer-name
+    exporter.clear()
+
+
+@pytest.fixture(autouse=True)
+def vertexai_init(vcr: VCR) -> None:
+    # Unfortunately I couldn't find a nice way to globally reset the global_config for each
+    # test because different vertex submodules reference the global instance directly
```

Comment:
mind opening a github issue on this in the google project, so we can link it and click there to see the details?

Reply:
To be clear this isn't a bug or anything. It's just difficult to monkey patch. This more of a note to self during development, I'd prefer to remove this comment as it's a bit distracting